Bio/Align/interfaces.py

```diff
@@ -142,7 +142,7 @@ class AlignmentWriter(ABC):
     """
 
     mode = "w"  # assume text files by default
-    fmt = None  # to be defined in the subclass
+    fmt: str  # to be defined in the subclass
```

Comment:
Here and above, previously the attribute was initialised as None. Now it is not initialised until the subclass does it (and it will always be a string)?

Reply:
As far as I understand, the subclass must define this variable. Whatever value we assign here would not be used anyway, so I decided to leave it uninitialized. We could assign an (empty) string which would be acceptable by mypy.

Without the change we get 14 errors like this:
```
Bio/Align/stockholm.py: note: In class "AlignmentWriter":
Bio/Align/stockholm.py:452:11: error: Incompatible types in assignment (expression has type "str", base class "AlignmentWriter" defined the type as "None")  [assignment]
        fmt = "Stockholm"
```

Side note: Type annotations without initialization is quite common with [dataclass](https://docs.python.org/3/library/dataclasses.html).

Reply:
> the subclass **must** define this variable

I have to correct myself: The subclass _may_ define this variable. The class `Bio.Align.bed.AlignmentWriter` currently does not define it. Under those circumstances, how about `fmt = "<undefined>"`? Or we go with `fmt: Optional[str] = None`

Reply:
Reverted to default `None`, with `Optional[str]` type. It's not my intention to change any behavior with this PR.

Reply:
No code change is simplest, although it is perhaps worth exploring why that subclass doesn't defined this.